src/vec3.h

```diff
@@ -1,35 +1,38 @@
 #include <cmath>
-#include <stdlib.h>
+#include <array>
+#include <iostream>
+
 
 class Vec3
 {
+
 private:
-    float v[3];
+    std::array<float, 3> v;
 
 public:
     Vec3(float x, float y, float z): v{x, y, z} {}
     Vec3(): v{0.0f, 0.0f, 0.0f} {}
 
-    float x() { return v[0]; }
-    float y() { return v[1]; }
-    float z() { return v[2]; }
+    float x() const { return v[0]; }
+    float y() const { return v[1]; }
+    float z() const { return v[2]; }
 	
 	inline const Vec3& operator+() const { return *this; }
 	inline Vec3 operator-() const { return Vec3(-v[0], -v[1], -v[2]); }
 	inline float operator[](int i) const { return v[i]; }
 	inline float& operator[](int i) { return v[i]; }
 	
-	inline Vec3& operator+=(const Vec3 &v2);
-	inline Vec3& operator-=(const Vec3 &v2);
-	inline Vec3& operator*=(const Vec3 &v2);
-	inline Vec3& operator/=(const Vec3 &v2);
+	inline Vec3& operator+=(const Vec3 &v);

```

Comment:
Non è un errore in senso classico, ma CLion mi notificava che nella dichiarazione la variabile si chiamava `v2` mentre nell'implementazione si chiamava `v` e lo flaggava come warning.

Reply:
Allora per uniformare lo chiamerei v2 come le altre e lo cambierei nell'implementazione

Reply:
Ti ci metti te su questi fix di `Vec3`?